pkg/api/types.go

```diff
@@ -476,6 +478,18 @@ type GlusterfsVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// MetadataVolumeSource represents an host directory mounted on a POD. It stores pod's metadata info
+type MetadataVolumeSource struct {
+	// PodNameFileName is the name of the file where name of the pod is written
+	PodNameFileName string `json:"podNameFileName,omitempty"`
```

Comment:
@thockin What you wrote is way better than what's in the PR at the moment, but it makes me scratch my head a bit; why not just write the entire pod to disk?